embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java

```diff
@@ -210,7 +216,29 @@ protected Schema setupTask(JdbcInputConnection con, PluginTask task) throws SQLE
         // build SELECT query and gets schema of its result
         String rawQuery = getRawQuery(task, con);
 
-        JdbcSchema querySchema = con.getSchemaOfQuery(rawQuery);
+        JdbcSchema querySchema = null;
+        if (task.getUseRawQueryWithIncremental()) {
+            String temporaryQuery = rawQuery;
+
+            // Insert pair of columnName:columnIndex order by column name length DESC
+            TreeMap<String, Integer> columnNames = new TreeMap<>(new Comparator<String>() {
+                @Override
+                public int compare(String val1, String val2) {
+                    return val2.length() - val1.length();
+                }
+            });
+            for (int i = 0; i < task.getIncrementalColumns().size(); i++) {
+                columnNames.put(task.getIncrementalColumns().get(i), i);
+            }
+
+            for (Map.Entry<String, Integer> column : columnNames.entrySet()) {
+                // Temporary replace place holder like ":id" with "?" to avoid SyntaxException while getting schema.
+                temporaryQuery = temporaryQuery.replaceAll(":" + column.getKey(), "?");
```

Comment:
We should use `replace` instead of `replaceAll`.

Reply:
Fixed at dbc6e380269aae3559528aa6d82f21c18a452619